observability/micrometer/src/main/java/org/wildfly/extension/micrometer/prometheus/PrometheusRegistryDefinitionRegistrar.java

```diff
@@ -0,0 +1,94 @@
+
+public class PrometheusRegistryDefinitionRegistrar implements ChildResourceDefinitionRegistrar, ResourceServiceConfigurator {
+    static final String NAME = "prometheus";
+    public static final PathElement PATH = PathElement.pathElement("registry", NAME);
+
+    static final String HTTP_EXTENSIBILITY_CAPABILITY = "org.wildfly.management.http.extensible";
+    static final SimpleAttributeDefinition CONTEXT = SimpleAttributeDefinitionBuilder
+            .create(MicrometerConfigurationConstants.CONTEXT, ModelType.STRING)
+            .setRequired(true)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+    static final AttributeDefinition SECURITY_ENABLED = SimpleAttributeDefinitionBuilder
+            .create(MicrometerConfigurationConstants.SECURITY_ENABLED, ModelType.BOOLEAN)
+            .setDefaultValue(ModelNode.FALSE)
+            .setRequired(false)
+            .setRestartAllServices()
+            .setAllowExpression(true)
+            .build();
+
+    public static final Collection<AttributeDefinition> ATTRIBUTES = List.of(CONTEXT, SECURITY_ENABLED);
+    private final WildFlyCompositeRegistry wildFlyRegistry;
+
+    public PrometheusRegistryDefinitionRegistrar(WildFlyCompositeRegistry wildFlyRegistry) {
+        this.wildFlyRegistry = wildFlyRegistry;
+    }
+
+    @Override
+    public ManagementResourceRegistration register(ManagementResourceRegistration parent, ManagementResourceRegistrationContext context) {
+        ResourceRegistration registration = ResourceRegistration.of(PATH);
+
+        ResourceDescriptor descriptor = ResourceDescriptor.builder(MicrometerSubsystemRegistrar.RESOLVER.createChildResolver(PATH))
+                .addAttributes(ATTRIBUTES)
+                .withRuntimeHandler(ResourceOperationRuntimeHandler.configureService(this))
+                .build();
+        ManagementResourceRegistration mrr = parent.registerSubModel(
+                ResourceDefinition.builder(registration, descriptor.getResourceDescriptionResolver()).build());
+        ManagementResourceRegistrar.of(descriptor).register(mrr);
+
+        return mrr;
+    }
+
+//    public Stability getStability() {
+//        return Stability.COMMUNITY;
+//    }
+
+    @Override
+    public ResourceServiceInstaller configure(OperationContext context, ModelNode model) throws OperationFailedException {
+        String serviceContext = CONTEXT.resolveModelAttribute(context, model).asStringOrNull();
+        boolean securityEnabled = SECURITY_ENABLED.resolveModelAttribute(context, model).asBoolean();
+
+        return ServiceInstaller.builder(ServiceDependency.on(HTTP_EXTENSIBILITY_CAPABILITY, ExtensibleHttpManagement.class))
+                .onStart(ehm -> {
+                    WildFlyPrometheusRegistry prometheusRegistry = new WildFlyPrometheusRegistry();
+                    wildFlyRegistry.add(prometheusRegistry);
+                    ehm.addManagementHandler(serviceContext, securityEnabled,
```

Comment:
Should this service also remove the management handler on stop?  Perhaps it doesn't matter since the service is active?

Reply:
Any adds/removes require a restart/reload. Does that answer your questions here?
```
                .withAddOperationRestartFlag(OperationEntry.Flag.RESTART_ALL_SERVICES)
                .withRemoveOperationRestartFlag(OperationEntry.Flag.RESTART_ALL_SERVICES)
```